micro-rdk-server/.cargo/config.toml

```diff
@@ -18,9 +18,3 @@ rustflags = ["-C", "default-linker-libraries"]
 linker = "ldproxy"
 runner = "espflash --monitor"
 rustflags = ["-C", "default-linker-libraries"]
-
-
-[env]
```

Comment:
These were the controlling values since `micro-rdk-server` is set as the default workspace member. However, I'm not convinced that `ESP_IDF_VERSION` had any effect since `ESP_IDF_TOOLS_INSTALL_DIR` was `fromenv`, and it would have just used whatever version was injected with `IDF_PATH/export.sh`.

The rule from https://github.com/esp-rs/esp-idf-sys/blob/master/BUILD-OPTIONS.md#esp-idf-configuration is that environment variable values (either in the shell or via an `[env]` section in `config.toml`) dominate. So the way we were doing it meant that they could not be overridden.

By moving to using the `Cargo.toml` section to set the defaults, we both let embuild/cargo/esp-idf worry about getting the right ESP-IDF version in place and we allow users to override the values on the command line. The canon shell takes advantage of that by using `ESP_IDF_TOOLS_INSTALL_DIR=fromenv` to honor the local shell environment, and developers can take advantage of it to opt into newer ESP-IDF versions trivially: `ESP_IDF_VERSION=4.4.8 make build-esp32-bin` will now use that version.